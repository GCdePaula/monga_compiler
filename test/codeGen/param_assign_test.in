

foo(i : int, j : int) {
  @i; @j;
  i = 10; @i;
  j = 11; @j;
  i = j; @i;

  return;
}

main() : int {
  foo(1, 2);
  return 0;
}
