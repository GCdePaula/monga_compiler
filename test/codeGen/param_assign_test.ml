open Base

let main () =
  let fname = "param_assign_test.monga" in
  Gen.test fname

let _ = Backtrace.Exn.with_recording true ~f:main
