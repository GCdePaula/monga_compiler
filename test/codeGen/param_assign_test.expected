FILE param_assign_test.monga LLVM CODE
; ModuleID = 'program'
source_filename = "program"

@0 = private unnamed_addr constant [5 x i8] c"%ld\0A\00", align 1

declare i32 @printf(i8*, ...)

declare i8* @malloc(i64)

define void @foo(i64, i64) {
entry:
  %2 = alloca i64
  store i64 %0, i64* %2
  %3 = alloca i64
  store i64 %1, i64* %3
  %4 = load i64, i64* %2
  %5 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0), i64 %4)
  %6 = load i64, i64* %3
  %7 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0), i64 %6)
  store i64 10, i64* %2
  %8 = load i64, i64* %2
  %9 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0), i64 %8)
  store i64 11, i64* %3
  %10 = load i64, i64* %3
  %11 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0), i64 %10)
  %12 = load i64, i64* %3
  store i64 %12, i64* %2
  %13 = load i64, i64* %2
  %14 = call i32 (i8*, ...) @printf(i8* getelementptr inbounds ([5 x i8], [5 x i8]* @0, i32 0, i32 0), i64 %13)
  ret void
}

define i64 @main() {
entry:
  call void @foo(i64 1, i64 2)
  ret i64 0
}

RETURN 0

OUTPUT
1
2
10
11
11
